lib/bolt/cli.rb

```diff
@@ -309,6 +315,9 @@ def parse
 
       validate(options)
 
+      # After validation, initialize inventory and targets. Errors here are better to catch early.
+      options[:targets] = inventory.get_targets(options[:nodes]) if options[:nodes]
```

Comment:
Partially it was to make it easier to test (see commit message). Otherwise it probably doesn't make much difference whether it's at the end of `parse` or beginning of `execute`.